accounts/abi/abi.go

```diff
@@ -149,6 +180,45 @@ func (abi ABI) Call(executer Executer, name string, args ...interface{}) interfa
 	return ret
 }
 
+func (abi ABI) UnmarshalCall(executer Executer, v interface{}, name string, args ...interface{}) error {
+	callData, err := abi.Pack(name, args...)
+	if err != nil {
+		return err
+	}
+
+	return abi.parseStruct(v, name, executer(callData))
+}
+
+func (abi ABI) parseStruct(v interface{}, name string, output []byte) error {
+	var method = abi.Methods[name]
+
+	if len(output) == 0 {
+		return fmt.Errorf("no output")
+	}
+
+	value := reflect.ValueOf(v).Elem()
+	typ := value.Type()
+
+	for i := 0; i < len(method.Outputs); i++ {
+		reflectValue := reflect.ValueOf(toGoType(i, method.Outputs[i], output))
+
+		for j := 0; j < typ.NumField(); j++ {
+			field := typ.Field(j)
+			// TODO read tags: `abi:"fieldName"`
+			if field.Name == strings.ToUpper(string(method.Outputs[i].Name[0]))+method.Outputs[i].Name[1:] {
```

Comment:
`strings.ToUpper(method.Outputs[i].Name[:1]) + method.Outputs[i].Name[1:]`

Bit cleaner, it doesn't need the extra char->string cast.